src/Networking/NexusMods.Networking.Downloaders/HttpDownloadJobWorker.cs

```diff
@@ -0,0 +1,51 @@
+using NexusMods.Abstractions.HttpDownloader;
+using NexusMods.Abstractions.Jobs;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.Paths;
+
+namespace NexusMods.Networking.Downloaders;
+
+/// <summary>
+/// A worker for downloading files from the internet.
+/// </summary>
+public class HttpDownloadJobWorker : APersistedJobWorker<HttpDownloadJob>
+{
+    private readonly IHttpDownloader _downloader;
+    private readonly IConnection _connection;
+
+    /// <summary>
+    /// DI constructor.
+    /// </summary>
+    /// <param name="connection"></param>
+    /// <param name="downloader"></param>
+    public HttpDownloadJobWorker(IConnection connection, IHttpDownloader downloader)
+    {
+        _connection = connection;
+        _downloader = downloader;
+    }
+
+    /// <inheritdoc />
+    protected override async Task<JobResult> ExecuteAsync(HttpDownloadJob job, CancellationToken cancellationToken)
+    {
+        var destination = job.Get(HttpDownloadJobPersistedState.Destination);
```

Comment:
Should be `job.Destination`. The job would have a property `Destination` like this:

```csharp
AbsolutePath Destination => Get(HttpDownloadJobPersistedState.Destination);
```

(same for `Uri` property)